content/en/blog/_posts/2022-10-30-forensic-container-checkpointing/index.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: blog
+title: "Kubernetes 1.25: alpha support for forensic container checkpointing"
+date: 2022-10-30
+slug: forensic-container-checkpointing-alpha
+---
+
+**Authors:** Adrian Reber (Red Hat)
+
+Kubernetes v1.25 introduces support for forensic container checkpointing.
+
+Forensic container checkpointing allows the creation of stateful copies of a
+running container without the container knowing that it is being checkpointed.
+The copy of the container can be analyzed and restored in a sandbox environment
+multiple times without the original container being aware of it.
+
+## How does it work?
+
+Forensic container checkpointing is based on
+[Checkpoint/Restore In Userspace(CRIU)][criu]. With the
+help of CRIU it is possible to checkpoint and restore containers.
+CRIU is integrated in runc, crun, CRI-O and containerd and
+forensic container checkpointing as implemented in Kubernetes
+uses these existing CRIU integrations.
+
+[criu]: https://criu.org/
+
+## Why is it important?
+
+With the help of CRIU and the corresponding integrations it is possible to get
+all information and state about a running container on disk for later forensic
+analysis. Forensic analysis might be important to inspect a suspicious
+container without stopping or influencing it. If the container is really under
+attack, the attacker might detect attempts to inspect the container. Taking a
+checkpoint and analysing the container in a sandboxed environment offers the
+possibility to inspect the container without the original container and maybe
+attacker being aware of the inspection.
+
+In addition to the forensic container checkpointing use case, it is also
+possible to migrate a container from one node to another node without loosing
+the internal state. Especially for stateful containers with long initialization
+times restoring from a checkpoint might save time after a reboot or enable much
+faster startup times.
+
+## How do I use forensic container checkpointing
+
+The feature is behind a feature gate, so make sure to enable the
+`ContainerCheckpointRestore` gate before you can use the new feature.
+
+The runtime must also support forensic container checkpointing:
+
+* containerd: support is currently under discussion. See containerd
+  pull request [#6965][containerd-checkpoint-restore-pr] for more details.
+
+* CRI-O: v1.25 has support for forensic container checkpointing.
+
+[containerd-checkpoint-restore-pr]: https://github.com/containerd/containerd/pull/6965
+
+### Usage example with CRI-O
+
+To use forensic container checkpointing in combination with CRI-O, CRI-O needs
+to be started with the command-line options `--enable-criu-support=true
+--drop-infra-ctr=false` and Kubernetes needs to be started with the
+`ContainerCheckpointRestore` feature gate enabled.
```

Comment:
I assume that you also need a low-level runtime (eg `runc`) that supports checkpointing,
and to have the `criu` tool installed on the same node. Is that right?

Reply:
added a sentence that it needs to be installed